lib/irb/completion.rb

```diff
@@ -111,8 +113,17 @@ def inspect
     end
 
     def completion_candidates(preposing, target, _postposing, bind:)
-      commands = command_completions(preposing, target)
+      preposing ||= ''
```

Comment:
@tompng Do you know when `proposing` will be `nil`? It looks like `ReplTypeCompletor` simply [assumes](https://github.com/ruby/repl_type_completor/blob/4e273330181bf096f71016937f52cddacfa77b7a/lib/repl_type_completor.rb#L33) it'd always be present?

Reply:
I think it wont be nil anymore.
In the old code, preposing and postposing might be nil
```ruby
IRB::InputCompletor::CompletionProc = ->(target, preposing=nil, postposing=nil){}
Readline.completion_proc = IRB::InputCompletor::CompletionProc # preposing and postposing are always nil
Reline.completion_proc = IRB::InputCompletor::CompletionProc # preposing and postposing are always string
```
But now, empty string is passed to `completion_candidates` in such case.
```ruby
Readline.completion_proc = ->(target) {
  bind = IRB.conf[:MAIN_CONTEXT].workspace.binding
  @completor.completion_candidates('', target, '', bind: bind)
}
```

In the last few lines in this file, `InputCompletor::CompletionProc` is the only place where nil might be passed to `RegexpCompletor#completion_candidates`, but it is not used from IRB's code.
```ruby
module InputCompletor
  CompletionProc = ->(target, preposing = nil, postposing = nil) {
    regexp_completor.completion_candidates(preposing, target, postposing, bind: IRB.conf[:MAIN_CONTEXT].workspace.binding)
   # maybe `(preposing || '', target, postposing || '', bind: ...)` is better
  }
end
deprecate_constant :InputCompletor
```

Reply:
I'll change `CompletionProc` then, thanks!